custom/enikshay/two_b_release_1/management/commands/enikshay_2b_case_properties.py

```diff
@@ -0,0 +1,496 @@
+            props['treatment_status'] = 'initiated_second_line_treatment'
+        elif treatment_status == 'yes_phi':
+            props['treatment_status'] = 'initiated_first_line_treatment'
+        elif treatment_status == 'yes_private':
+            props['treatment_status'] = 'initiated_outside_facility'
+
+        if treatment_status == 'yes_phi':
+            props['treatment_initiated'] = 'yes_phi'
+        elif treatment_status == 'yes_private':
+            props['treatment_initiated'] = 'yes_private'
+        elif treatment_status:
+            props['treatment_initiated'] = 'no'
+
+        if not episode.get_case_property('date_of_diagnosis'):
+            props['date_of_diagnosis'] = episode.get_case_property('date_reported')
+
+        return CaseStructure(
+            case_id=episode.case_id,
+            walk_related=False,
+            attrs={
+                "create": False,
+                "update": props,
+            },
+        )
+
+    def migrate_test(self, test, person):
+        self.total_tests += 1
+        props = {
+            'is_direct_test_entry': 'no',
+            'rft_drtb_diagnosis': test.get_case_property('diagnostic_drtb_test_reason'),
+            'dataset': person.get_case_property('dataset'),
+        }
+
+        if test.get_case_property('follow_up_test_reason') == 'private_ntm':
+            props['rft_general'] = 'diagnosis_dstb'
+            props['rft_dstb_diagnosis'] = 'private_ntm'
+        else:
+            props['rft_general'] = {
+                'diagnostic': 'diagnosis_dstb',
+                'diagnosis_dstb': 'diagnosis_dstb',
+                'follow_up_dstb': 'follow_up_dstb',
+                'diagnosis_drtb': 'diagnosis_drtb',
+            }.get(test.get_case_property('purpose_of_testing'), "")
+            props['rft_dstb_diagnosis'] = test.get_case_property('diagnostic_test_reason')
+            props['rft_dstb_followup'] = test.get_case_property('follow_up_test_reason')
+
+        return CaseStructure(
+            case_id=test.case_id,
+            walk_related=False,
+            attrs={
+                "create": False,
+                "update": props,
+            },
+        )
+
+    def migrate_referral(self, referral, occurrences):
+        self.total_referrals += 1
+        prop = referral.get_case_property
+        props = {
+            'referral_initiated_date': (prop('referral_date') or prop('date_of_referral')),
+            'referred_to_name': prop('referred_to_location_name'),
+            'referred_by_name': prop('referred_by'),
+            'referral_rejection_reason_other_detail': prop('reason_for_refusal_other_detail'),
+            'referral_rejection_reason': prop('reason_for_refusal'),
+            'referral_closed_date': prop('acceptance_refusal_date'),
+            'accepted_by_name': prop('phi'),
+        }
+
+        if occurrences:
+            occurrence = max((case.opened_on, case) for case in occurrences)[1]
```

Comment:
looks like you calculate this for each referral and each trail. might be cacheable somewhere?